### PR TITLE
✨ Source Snapchat Marketing : add attribution_windows and action_report_time to request parameters

### PR DESCRIPTION
**Pull request by Abuton**
## What
*Describe what the change is solving*
image

Adding the following parameters as Optional to the snapchat-marketing source connector
- action_report_time
- swipe_up_attribution_window
- view_attribution_window

These parameters as the default value has defined in the [snapchat documentation](https://marketingapi.snapchat.com/docs/#request-response-pattern)
*It helps to add screenshots if it affects the frontend.*

## How
I have added these parameters to the `source.py` and set the default values as defined on snapchat documentation. I have done this in other to allow users to have different kind of report from the API based on their defined values for these parameters.

## Recommended reading order
1. `source.py`
2. `unit_test.oy`

## User Impact

This change will not break any already existing connections to this source as the new parameters are optional with default value same as what already exist before.

The result of this PR will allow users to have report based on their accounts defined parameters. With the addition of `action_report_time` parameters, users will be able to select between impression or conversion kind of report.

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added
image

- Issue acceptance criteria met
- PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests#pull-request-title-convention)
- If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:generateScaffolds` then checking in your changes
- Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-snapchat-marketing/source_snapchat_marketing/source.py [hunk 1]
```diff
@@ -808,6 +814,9 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
             ),
             "start_date": config["start_date"],
             "end_date": config.get("end_date", default_end_date),
+            "action_report_time": config.get("action_report_time", "impression"),
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-snapchat-marketing/source_snapchat_marketing/source.py:817]: Default should be `conversion`.

File: airbyte-integrations/connectors/source-snapchat-marketing/source_snapchat_marketing/source.py [hunk 2]
```diff
@@ -808,6 +814,9 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
...
+            "swipe_up_attribution_window": config.get("swipe_up_attribution_window", "28_DAY"),
+            "view_attribution_window": config.get("view_attribution_window", "7_DAY"),
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-snapchat-marketing/source_snapchat_marketing/source.py:819]: Default should be `1_DAY`.

File: airbyte-integrations/connectors/source-snapchat-marketing/source_snapchat_marketing/spec.json [hunk 3]
```diff
@@ -45,6 +45,30 @@
         "examples": ["2022-01-30"],
         "order": 4,
         "format": "date"
+      },
+      "action_report_time": {
+        "type": "string",
+        "enum": ["conversion", "impression"],
+        "title": "Action Report Time",
+        "description": "Specifies the principle for conversion reporting",
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-snapchat-marketing/source_snapchat_marketing/spec.json:53]: Please add dots at the end of descriptions.

File: docs/integrations/sources/snapchat-marketing.md [hunk 4]
```diff
@@ -17,6 +17,10 @@ This page guides you through the process of setting up the Snapchat Marketing so
 * client_secret
 * refresh_token
 * start_date
+* end_date
+* action_report_time (Optional)
```
Comment (lazebnyi) [inline docs/integrations/sources/snapchat-marketing.md:21]: Please add more details for what those param responsible for and what is default value for them, even it's empty.

**Review by lazebnyi — CHANGES_REQUESTED** [review]

File: docs/integrations/sources/snapchat-marketing.md (hunk 4 above)
Comment (Abuton) [reply docs/integrations/sources/snapchat-marketing.md]: Done

File: airbyte-integrations/connectors/source-snapchat-marketing/source_snapchat_marketing/spec.json (hunk 3 above)
Comment (Abuton) [reply airbyte-integrations/connectors/source-snapchat-marketing/source_snapchat_marketing/spec.json]: Done

File: airbyte-integrations/connectors/source-snapchat-marketing/source_snapchat_marketing/source.py (hunk 2 above)
Comment (Abuton) [reply airbyte-integrations/connectors/source-snapchat-marketing/source_snapchat_marketing/source.py]: Changed

**Review by lazebnyi — APPROVED** [review]
lgtm

**Pull request merged**